builder/amazon: Chroot regresion

The following template works with 0.10.1 but crashes on master:

```json
{
  "builders": [{
    "type": "amazon-chroot",
    "region": "eu-central-1",
    "ami_virtualization_type": "hvm",
    "source_ami": "ami-eab64b85",
    "ami_name": "packer-example {{timestamp}} coreos",
    "root_volume_size": 8,
    "mount_partition": 9,
    "chroot_mounts": [
         [ "proc", "proc", "/proc" ],
         [ "sysfs", "sysfs", "/sys" ],
         [ "bind", "/dev", "/dev" ],
         [ "devpts", "devpts", "/dev/pts" ]
    ]
  }]
}
```

```
==> amazon-chroot: Deleting the created EBS volume...
Build 'amazon-chroot' errored: unexpected EOF

==> Some builds didn't complete successfully and had errors:
--> amazon-chroot: unexpected EOF

==> Builds finished but no artifacts were created.
panic: runtime error: invalid memory address or nil pointer dereference
2016/09/16 23:06:20 packer: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5e3603]
2016/09/16 23:06:20 packer:
2016/09/16 23:06:20 packer: goroutine 56 [running]:
2016/09/16 23:06:20 packer: panic(0xfdd5a0, 0xc42000c0d0)
2016/09/16 23:06:20 packer:     /opt/boxen/homebrew/Cellar/go/1.7/libexec/src/runtime/panic.go:500 +0x1a1
2016/09/16 23:06:20 packer: github.com/mitchellh/packer/builder/amazon/chroot.(*StepRegisterAMI).Run(0xc420298520, 0x17d4c80, 0xc420293020, 0x0)
2016/09/16 23:06:20 packer:     /Users/rickard/src/go-code/src/github.com/mitchellh/packer/builder/amazon/chroot/step_register_ami.go:43 +0x2b3
2016/09/16 23:06:20 packer: github.com/mitchellh/packer/vendor/github.com/mitchellh/multistep.(*BasicRunner).Run(0xc420251a80, 0x17d4c80, 0xc420293020)
2016/09/16 23:06:20 packer:     /Users/rickard/src/go-code/src/github.com/mitchellh/packer/vendor/github.com/mitchellh/multistep/basic_runner.go:70 +0x207
2016/09/16 23:06:20 packer: github.com/mitchellh/packer/builder/amazon/chroot.(*Builder).Run(0xc4200a9340, 0x17d7480, 0xc420229bc0, 0x17d0b40, 0xc420298440, 0x17d6240, 0xc4200224c0, 0xc420292a20, 0x0, 0x0, ...)
2016/09/16 23:06:20 packer:     /Users/rickard/src/go-code/src/github.com/mitchellh/packer/builder/amazon/chroot/builder.go:274 +0xd24
2016/09/16 23:06:20 packer: github.com/mitchellh/packer/packer/rpc.(*BuilderServer).Run(0xc420228220, 0x1, 0xc4202983a0, 0x0, 0x0)
2016/09/16 23:06:20 packer:     /Users/rickard/src/go-code/src/github.com/mitchellh/packer/packer/rpc/builder.go:93 +0x218
2016/09/16 23:06:20 packer: reflect.Value.call(0xc420015e00, 0xc420022418, 0x13, 0x1188c2b, 0x4, 0xc4200b9ed0, 0x3, 0x3, 0x0, 0x0, ...)
2016/09/16 23:06:20 packer:     /opt/boxen/homebrew/Cellar/go/1.7/libexec/src/reflect/value.go:434 +0x5c8
2016/09/16 23:06:20 packer: reflect.Value.Call(0xc420015e00, 0xc420022418, 0x13, 0xc4200b9ed0, 0x3, 0x3, 0xc42024f4f8, 0x1, 0x0)
2016/09/16 23:06:20 packer:     /opt/boxen/homebrew/Cellar/go/1.7/libexec/src/reflect/value.go:302 +0xa4
2016/09/16 23:06:20 packer: net/rpc.(*service).call(0xc42011be00, 0xc42011bdc0, 0xc420216be0, 0xc420063980, 0xc4202285c0, 0xf559a0, 0xc42029839c, 0x18a, 0xf34c80, 0xc4202983a0, ...)
2016/09/16 23:06:20 packer:     /opt/boxen/homebrew/Cellar/go/1.7/libexec/src/net/rpc/server.go:383 +0x148
2016/09/16 23:06:20 packer: created by net/rpc.(*Server).ServeCodec
2016/09/16 23:06:20 packer:     /opt/boxen/homebrew/Cellar/go/1.7/libexec/src/net/rpc/server.go:477 +0x421
2016/09/16 23:06:20 ui error: Build 'amazon-chroot' errored: unexpected EOF
2016/09/16 23:06:20 Builds completed. Waiting on interrupt barrier...
2016/09/16 23:06:20 machine readable: error-count []string{"1"}
```

--- a/builder/amazon/chroot/step_register_ami.go
+++ b/builder/amazon/chroot/step_register_ami.go
@@ -25,20 +25,22 @@ func (s *StepRegisterAMI) Run(state multistep.StateBag) multistep.StepAction {
 
 	var (
 		registerOpts   *ec2.RegisterImageInput
-		blockDevices   []*ec2.BlockDeviceMapping
+		mappings       []*ec2.BlockDeviceMapping
 		image          *ec2.Image
 		rootDeviceName string
 	)
 
 	if config.FromScratch {
-		blockDevices = config.AMIBlockDevices.BuildAMIDevices()
+		mappings = config.AMIBlockDevices.BuildAMIDevices()
 		rootDeviceName = config.RootDeviceName
 	} else {
 		image = state.Get("source_image").(*ec2.Image)
-		blockDevices = make([]*ec2.BlockDeviceMapping, len(image.BlockDeviceMappings))
+		mappings = image.BlockDeviceMappings
 		rootDeviceName = *image.RootDeviceName
 	}
-	for i, device := range blockDevices {
+
+	newMappings := make([]*ec2.BlockDeviceMapping, len(mappings))
+	for i, device := range mappings {
 		newDevice := device
 		if *newDevice.DeviceName == rootDeviceName {
 			if newDevice.Ebs != nil {
@@ -58,7 +60,7 @@ func (s *StepRegisterAMI) Run(state multistep.StateBag) multistep.StepAction {
 			newDevice.Ebs.Encrypted = nil
 		}
 
-		blockDevices[i] = newDevice
+		newMappings[i] = newDevice
 	}
 
 	if config.FromScratch {
@@ -67,10 +69,10 @@ func (s *StepRegisterAMI) Run(state multistep.StateBag) multistep.StepAction {
 			Architecture:        aws.String(ec2.ArchitectureValuesX8664),
 			RootDeviceName:      aws.String(rootDeviceName),
 			VirtualizationType:  aws.String(config.AMIVirtType),
-			BlockDeviceMappings: blockDevices,
+			BlockDeviceMappings: newMappings,
 		}
 	} else {
-		registerOpts = buildRegisterOpts(config, image, blockDevices)
+		registerOpts = buildRegisterOpts(config, image, newMappings)
 	}
 
 	// Set SriovNetSupport to "simple". See http://goo.gl/icuXh5
@@ -112,12 +114,12 @@ func (s *StepRegisterAMI) Run(state multistep.StateBag) multistep.StepAction {
 
 func (s *StepRegisterAMI) Cleanup(state multistep.StateBag) {}
 
-func buildRegisterOpts(config *Config, image *ec2.Image, blockDevices []*ec2.BlockDeviceMapping) *ec2.RegisterImageInput {
+func buildRegisterOpts(config *Config, image *ec2.Image, mappings []*ec2.BlockDeviceMapping) *ec2.RegisterImageInput {
 	registerOpts := &ec2.RegisterImageInput{
 		Name:                &config.AMIName,
 		Architecture:        image.Architecture,
 		RootDeviceName:      image.RootDeviceName,
-		BlockDeviceMappings: blockDevices,
+		BlockDeviceMappings: mappings,
 		VirtualizationType:  image.VirtualizationType,
 	}
 

